src/typeInference/annotateAST.ts

```diff
@@ -267,6 +268,13 @@ export function annotate(ast: IAST, staticContext: StaticContext): SequenceType
 			);
 			annotateTypeSwitchOperator(ast);
 
+		case 'quantifiedExpr':
+			// Variable amount of children from this AST node, solution: looping through them
+			for (let iterationCounter = 1; iterationCounter < ast.length; iterationCounter++) {
```

Comment:
Maybe you could use astHelper.getChildren(ast, '*') for this one and then loop with a map. That would maybe be cleaner.
astHelper.getChildren(ast, '*').map((a) => annotate(a, staticContext));

Reply:
Thanks that definitely looks a lot better indeed